tests/layers/tuple_matchers/slot_similarity_tuple_matcher_test.py

```diff
@@ -0,0 +1,114 @@
+        mask2 = K.variable(numpy.asarray([0, 0, 0]))
+        calculated_mask_exclude = K.eval(
+            SlotSimilarityTupleMatcher("cosine_similarity",
+                                       self.num_hidden_layers,
+                                       self.hidden_layer_width,
+                                       hidden_layer_activation=self.hidden_layer_activation)
+                .compute_mask([tuple1, tuple2], [mask1, mask2]))
+        assert_array_almost_equal(calculated_mask_exclude, numpy.array([0], dtype='int32'))
+
+        # Test when tuple2 is valid.
+        # Here, since tuple2 is valid, we want to return a mask value of 1 for this pair
+        mask2 = K.variable(numpy.asarray([0, 1, 0]))
+        calculated_mask_include = K.eval(
+            SlotSimilarityTupleMatcher("cosine_similarity",
+                                       self.num_hidden_layers,
+                                       self.hidden_layer_width,
+                                       hidden_layer_activation=self.hidden_layer_activation)
+                .compute_mask([tuple1, tuple2], [mask1, mask2]))
+        assert_array_almost_equal(calculated_mask_include, numpy.array([1], dtype='int32'))
+
+    def test_handles_input_masks_correctly(self):
+        mask_layer = Masking(mask_value=0.0)
+        masked_tuple1 = mask_layer(self.tuple1_input)
+        masked_tuple2 = mask_layer(self.tuple2_input)
+        # Add a set of paddings to slot 1 in tuple 2
+        self.tuple2[:, 1, :] = numpy.zeros(4)
+        match_layer = SlotSimilarityTupleMatcher("cosine_similarity",
+                                                 self.num_hidden_layers,
+                                                 self.hidden_layer_width,
+                                                 hidden_layer_activation=self.hidden_layer_activation)
+        output = match_layer([masked_tuple1, masked_tuple2])
+        mask_model = Model([self.tuple1_input, self.tuple2_input], output)
+
+        # Testing masked case
+        # Testing general unmasked case.
+        similarity_function = CosineSimilarity(name="cosine_similarity")
+        cosine_similarities = similarity_function.compute_similarity(K.variable(self.tuple1), K.variable(self.tuple2))
+        mask2 = K.variable(numpy.asarray([[1, 0, 1]], dtype='float32'))
+        masked_cosine_similarities = cosine_similarities * mask2
+
+        # Get the initial weights for use in testing
+        dense_hidden_weights = K.eval(mask_model.trainable_weights[0])
+        score_weights = K.eval(mask_model.trainable_weights[1])
+
+        # Desired_overlap gets fed into the inner NN.
+        dense1_activation = numpy.dot(K.eval(masked_cosine_similarities), dense_hidden_weights)
+        final_score = numpy.dot(dense1_activation, score_weights)
+        desired_result = logistic.cdf(final_score)
+        result = mask_model.predict([self.tuple1, self.tuple2])
+        assert_array_almost_equal(result, desired_result)
```

Comment:
This red icon at the end of the line means that your file doesn't end in a newline.  Can you fix that?